### DISCUSSION TITLE
How do I find out AES-128-CBC encryption key?

### DISCUSSION THREAD

**Discussion by puujee12** [rgerganov/gen5fw#3] [Q&A] [answered]
Hello @rgerganov and thank you for your amazing blog post. I have Gen5 head unit from Kona and it's firmware is pretty much same. So I will be very glad if you share your method on how did you find aes encryption key.

**Comment by rgerganov** [comment, marked as answer]
The encryption key and IV are the same for all cars using Gen5: 
```
AES_KEY = b')1Zorxo^fAhGlh$#'
AES_IV = b'aoAkfwk+#1.6G{dE'
```
I found them using Ghidra and the SDK for TCC893X

**Reply by puujee12** [reply]
Thank you very much for your reply. @rgerganov 
And if is it possible, could you kindly share SDK for TCC893X? I searched it for few days and i couldn't find the one with working link anywhere.

**Reply by rgerganov** [reply]
Contact me on email or twitter DM
